pkg/cfg/api.go

```diff
@@ -78,26 +98,35 @@ func DefaultAPIConfig() API {
 
 // API is carbonapi-specific config
 type API struct {
-	// TODO (grzkv): Why does carbonapi config refer to zipper config?
-	// It should probably refer to the common one
 	Zipper `yaml:",inline"`
 
-	// TODO (grzkv): Move backends list to a single backend here
-
-	ResolveGlobs                     int           `yaml:"resolveGlobs"`
-	EnableCacheForRenderResolveGlobs bool          `yaml:"enableCacheForRenderResolveGlobs"`
-	Cache                            CacheConfig   `yaml:"cache"`
-	TimezoneString                   string        `yaml:"tz"`
-	PidFile                          string        `yaml:"pidFile"`
-	BlockHeaderFile                  string        `yaml:"blockHeaderFile"`
-	BlockHeaderUpdatePeriod          time.Duration `yaml:"blockHeaderUpdatePeriod"`
-	HeadersToLog                     []string      `yaml:"headersToLog"`
-
-	UnicodeRangeTables        []string          `yaml:"unicodeRangeTables"`
-	IgnoreClientTimeout       bool              `yaml:"ignoreClientTimeout"`
-	DefaultColors             map[string]string `yaml:"defaultColors"`
-	FunctionsConfigs          map[string]string `yaml:"functionsConfig"`
-	GraphiteVersionForGrafana string            `yaml:"graphiteVersionForGrafana"`
+	ResolveGlobs                     int               `yaml:"resolveGlobs"`
+	EnableCacheForRenderResolveGlobs bool              `yaml:"enableCacheForRenderResolveGlobs"`
+	Cache                            CacheConfig       `yaml:"cache"`
+	TimezoneString                   string            `yaml:"tz"`
+	PidFile                          string            `yaml:"pidFile"`
+	BlockHeaderFile                  string            `yaml:"blockHeaderFile"`
+	BlockHeaderUpdatePeriod          time.Duration     `yaml:"blockHeaderUpdatePeriod"`
+	HeadersToLog                     []string          `yaml:"headersToLog"`
+	UnicodeRangeTables               []string          `yaml:"unicodeRangeTables"`
+	IgnoreClientTimeout              bool              `yaml:"ignoreClientTimeout"`
+	DefaultColors                    map[string]string `yaml:"defaultColors"`
+	FunctionsConfigs                 map[string]string `yaml:"functionsConfig"`
+	GraphiteVersionForGrafana        string            `yaml:"graphiteVersionForGrafana"`
+
+	// The size of the requests queue propagated to backends.
+	// During this stage of refactoring it is a placeholder and should not fill-up.
+	// At the later stages it will play a key role in the request processing.
+	QueueSize                     int `yaml:"queueSize"`
+	MaxConcurrentUpstreamRequests int `yaml:"maxConcurrentUpstreamRequests"`
+	// The number of workers to process requests queue.
+	ProcWorkers int `yaml:"procWorkers"`
+	// The threshold of the number of sub-requests after which the render requests are considered large.
+	// It is used to select the processing queue: Small requests get on the fast queue, large ones on the slow one.
```

Comment:
Queues consist of sub-requests. So, in order to make sure that the right message is transmitted, I propose:

> sub-requests derived from small requests get on the fast queue, and the ones derived from large requests get on the slow queue.

instead of:

> Small requests get on the fast queue, large ones on the slow one.

Reply:
I'm not sure I see it that way. This is a matter of terminology.
I'll try to make things clearer moving forward.